pkg/agent/core/ngt/handler/grpc/handler_test.go

```diff
@@ -5815,16 +5820,501 @@ func Test_server_MultiInsert(t *testing.T) {
+				args: args{
+					ctx:  ctx,
+					reqs: req,
+				},
+				fields: fields{
+					name:    name,
+					ip:      ip,
+					svcCfg:  defaultF32SvcCfg,
+					svcOpts: defaultSvcOpts,
+				},
+				beforeFunc: func(t *testing.T, s *server) {
+					vecs := genF32Vec(vector.Gaussian, 100, f32VecDim)
+					for i, r := range req.Requests {
+						ir := &payload.Insert_Request{
+							Vector: &payload.Object_Vector{
+								Id:     r.Vector.Id,
+								Vector: vecs[i],
+							},
+							Config: &payload.Insert_Config{
+								SkipStrictExistCheck: false,
+							},
+						}
+						if _, err := s.Insert(ctx, ir); err != nil {
+							t.Fatal(err)
+						}
+					}
+					if _, err := s.CreateIndex(ctx, &payload.Control_CreateIndexRequest{
+						PoolSize: uint32(len(req.Requests)),
+					}); err != nil {
+						t.Fatal(err)
+					}
+				},
+				want: want{
+					haveErr: true,
+				},
+			}
+		}(),
+		func() test {
+			req := genMultiInsertReq(Float, vector.Gaussian, 100, f32VecDim)
+
+			return test{
+				name: "Decision Table Testing case 4.3: Fail to MultiInsert with 2 existed ID when SkipStrictExistCheck is true",
+				args: args{
+					ctx:  ctx,
+					reqs: req,
+				},
+				fields: fields{
+					name:    name,
+					ip:      ip,
+					svcCfg:  defaultF32SvcCfg,
+					svcOpts: defaultSvcOpts,
+				},
+				beforeFunc: func(t *testing.T, s *server) {
+					vecs := genF32Vec(vector.Gaussian, 2, f32VecDim)
+					for i, j := 49, 0; i < 51; i, j = i+1, j+1 { // start from the middle of insert request
+						ir := &payload.Insert_Request{
+							Vector: &payload.Object_Vector{
+								Id:     req.Requests[i].Vector.Id,
+								Vector: vecs[j],
+							},
+							Config: &payload.Insert_Config{
+								SkipStrictExistCheck: true,
+							},
+						}
+						if _, err := s.Insert(ctx, ir); err != nil {
+							t.Fatal(err)
+						}
+					}
+					if _, err := s.CreateIndex(ctx, &payload.Control_CreateIndexRequest{
+						PoolSize: 2,
+					}); err != nil {
+						t.Fatal(err)
+					}
+				},
+				want: want{
+					haveErr: true,
+				},
+			}
+		}(),
+		func() test {
+			req := genMultiInsertReq(Float, vector.Gaussian, 100, f32VecDim)
+
+			return test{
+				name: "Decision Table Testing case 4.4: Fail to MultiInsert with all existed ID when SkipStrictExistCheck is true",
+				args: args{
+					ctx:  ctx,
+					reqs: req,
+				},
+				fields: fields{
+					name:    name,
+					ip:      ip,
+					svcCfg:  defaultF32SvcCfg,
+					svcOpts: defaultSvcOpts,
+				},
+				beforeFunc: func(t *testing.T, s *server) {
+					vecs := genF32Vec(vector.Gaussian, 100, f32VecDim)
+					for i, r := range req.Requests {
+						ir := &payload.Insert_Request{
+							Vector: &payload.Object_Vector{
+								Id:     r.Vector.Id,
+								Vector: vecs[i],
+							},
+							Config: &payload.Insert_Config{
+								SkipStrictExistCheck: false,
+							},
+						}
+						if _, err := s.Insert(ctx, ir); err != nil {
+							t.Fatal(err)
+						}
+					}
```

Comment:
```suggestion
					for i, r := range req.Requests {
						ir := r
						ir.Vector.Vector = vecs[i],
						ir.Config.SkipStrictExistCheck = true
						if _, err := s.Insert(ctx, ir); err != nil {
							t.Fatal(err)
						}
					}
```

Reply:
We need to create another insert request and copy the field to it, your suggestion will manipulate the content of `req`.

Reply:
It's ok, but you can update more simply.

Reply:
sorry but what do you mean by `update more simply`?

Reply:
IMO `ir := *r` is fine?

Reply:
tI wouldn't work. We may need to implement clone function for it, what do you think?